hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java

```diff
@@ -186,19 +177,26 @@ public void notifyLeaderChanged(RaftGroupMemberId groupMemberId,
    * @param index index which is being updated
    */
   @Override
-  public void notifyTermIndexUpdated(long currentTerm, long index) {
-    // SnapshotInfo should be updated when the term changes.
-    // The index here refers to the log entry index and the index in
-    // SnapshotInfo represents the snapshotIndex i.e. the index of the last
-    // transaction included in the snapshot. Hence, snaphsotInfo#index is not
-    // updated here.
+  public synchronized void notifyTermIndexUpdated(long currentTerm, long index) {
+    final TermIndex newTermIndex = TermIndex.valueOf(currentTerm, index);
+    lastNotifiedTermIndex = assertUpdateIncreasingly("lastNotified", lastNotifiedTermIndex, newTermIndex);
```

Comment:
I have seen issues while handling with earlier PR, that, leader election depends on  notifiedTermIndex (as updated to lastAppliedTermIndex), but here we are not updating to lastAppliedTermIndex.

Check if this can have impact to ratis, as this information is not known to ratis internally till snapshot is taken.

Reply:
No, leader election  has nothing to do with notified/applied indices.